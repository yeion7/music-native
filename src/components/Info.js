/* @flow weak */

import React from "react";
import { View } from "react-native";
import { Text } from "native-base";
import styled from "styled-components/native";

import { formattedTime } from "../utils/time";

const Info = ({ song, position, duration }) => (
  <Container>
    <Text
      style={{ textAlign: "center", paddingHorizontal: 10 }}
      ellipsizeMode={"tail"}
      numberOfLines={1}
    >
      {song.name}
    </Text>
    <Text note>
      {formattedTime(position)}
      {" "}
      /
      {" "}
      {formattedTime(duration)}
    </Text>
  </Container>
);

const Container = styled.View`
  align-items: center;
  justify-content: center;
  padding-bottom: 2;
`;

export default Info;
