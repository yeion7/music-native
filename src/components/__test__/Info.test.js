import React from "react";
import "react-native";
import renderer from "react-test-renderer";

import Info from "../Info";

describe("renders test ", () => {
  it("render Info component", () => {
    const tree = renderer
      .create(
        <Info song={{ name: "Aw Naw" }} position={40000} duration={10000} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
