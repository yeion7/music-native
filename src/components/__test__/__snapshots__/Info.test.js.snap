// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders test  render Info component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 2,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "color": "#000",
        "fontFamily": "System",
        "fontSize": 16,
        "paddingHorizontal": 10,
        "textAlign": "center",
      }
    }
    virtual={undefined}
  >
    Aw Naw
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    note={true}
    style={
      Object {
        "color": "#a7a7a7",
        "fontFamily": "System",
        "fontSize": 14,
      }
    }
    virtual={undefined}
  >
    0:40
     
    /
     
    0:10
  </Text>
</View>
`;
